website/content/docs/upgrading/deduplication/terraform-refs.mdx

```diff
@@ -0,0 +1,160 @@
+- You have admin permission on the relevant Vault server or cluster.
+
+</Tip>
+
+
+## How renaming affects external references
+
+Renaming entities and groups can break references in Terraform configuration
+files when those reference refer directly to the entity or group by name. For
+example, assume you have a Terraform configuration file with named identity
+resources like the following:
+
+<CodeBlockConfig highlight="11,16" hideClipboard="true">
+
+```hcl
+terraform {
+  required_providers {
+    vault = {
+      source = "hashicorp/vault"
+    }
+  }
+}
+
+provider "vault" {}
+
+resource "vault_identity_entity" "BOB" {
+  name      = "BOB"
+  policies  = ["TEST"]
+}
+
+resource "vault_identity_entity" "bob" {
+    name      = "bob"
+    policies  = ["test"]
+}
+```
+
+</CodeBlockConfig>
+
+By default, Vault ignore case when matching identities, treats `BOB` and `bob`
+as the same name, and rejects the second resource as a duplicate. However, if
+your Vault cluster is running in a mode that allows both resource names due to
+historical issues, the resources might exist as separate entities.
+
+If Vault identifies `bob` and `BOB` as duplicates during deduplication, it
+renames one of the identities `<name>-<uuid>`. After deduplication, Terraform
+tries to rename the related resource, but the in-place update will fail because
+the existing resource now violates the case-insensitive name constraint on the
+Vault side.
+
+For example:
+
+<CodeBlockConfig highlight="2,3,11,21,28" hideClipboard="true">
+
+```
+➜  tf_dupe_testing terraform apply
+vault_identity_entity.bob: Refreshing state... [id=e8c5e633-fe37-5a49-4a29-32e2643d03bd]
+vault_identity_entity.BOB: Refreshing state... [id=2577bc3f-67ab-dab7-93dc-e86f78194ff0]
+
+Terraform used the selected providers to generate the following execution plan. Resource
+actions are indicated with the following symbols:
+  ~ update in-place
+
+Terraform will perform the following actions:
+
+  # vault_identity_entity.bob will be updated in-place
+  ~ resource "vault_identity_entity" "bob" {
+      + external_policies = false
+        id                = "e8c5e633-fe37-5a49-4a29-32e2643d03bd"
+      ~ name              = "bob-e8c5e633-fe37-5a49-4a29-32e2643d03bd" -> "bob"
+        # (3 unchanged attributes hidden)
+    }
+
+...
+
+vault_identity_entity.bob: Modifying... [id=e8c5e633-fe37-5a49-4a29-32e2643d03bd]
+╷
+│ Error: error updating IdentityEntity "e8c5e633-fe37-5a49-4a29-32e2643d03bd": Error making API request.
+│
+│ URL: PUT https://127.0.0.1:8200/v1/identity/entity/id/e8c5e633-fe37-5a49-4a29-32e2643d03bd
+│ Code: 400. Errors:
+│
+│ * entity name is already in use
+│
+│   with vault_identity_entity.bob,
+│   on main.tf line 17, in resource "vault_identity_entity" "bob":
+│   17: resource "vault_identity_entity" "bob" {
+```
+
+</CodeBlockConfig>
+
+
+## Solution
+
+The easiest way to deal with renamed entities is to manually update the
```

Comment:
nit: this can impact both entities and groups. Not sure if we should call that out here just to be clear?

Reply:
fixed